addons/binding/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/internal/handler/FoobotHandler.java

```diff
@@ -0,0 +1,306 @@
+        }
+
+        if (devices == null) {
+            errorMsg = "No devices found on this account.";
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            return;
+        }
+
+        for (FoobotDevice ob : devices) {
+            // Compare the mac address to each record in order to fetch the UUID of the current device.
+            if (config.mac.equals(ob.getMac())) {
+                this.device = ob;
+                break;
+            }
+        }
+
+        if (this.device == null) {
+            errorMsg = "No device found for this MAC address.";
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+        startAutomaticRefresh();
+    }
+
+    /**
+     * Start the job refreshing the Sensor data from Foobot device
+     */
+    private void startAutomaticRefresh() {
+        int delay = (config.refreshIntervalInMinutes != null) ? config.refreshIntervalInMinutes.intValue()
+                : FoobotBindingConstants.DEFAULT_REFRESH_PERIOD_MINUTES;
+        refreshDataJob = scheduler.scheduleWithFixedDelay(() -> {
+            // Request new sensor data from foobot.io service
+            foobotData = updateFoobotData();
+
+            // Update all channels from the updated Sensor data
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId(), foobotData);
+            }
+        }, 0, delay, TimeUnit.MINUTES);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the Foobot handler.");
+        if (refreshDataJob != null && !refreshDataJob.isCancelled()) {
+            refreshDataJob.cancel(true);
+            refreshDataJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(@NonNullByDefault ChannelUID channelUID, @NonNullByDefault Command command) {
+        if (command instanceof RefreshType) {
+            return;
+        } else {
+            logger.debug("The Foobot binding is read-only and can not handle command {}", command);
+        }
+    }
+
+    private void updateChannel(String channelId, FoobotJsonData foobotResponse) {
+        Object value = null;
+        try {
+            value = getValue(channelId, foobotResponse);
+        } catch (RuntimeException rte) {
+            logger.debug("RuntimeException while getting Foobot sensor data for channel id = {}: {}",
+                    channelId.toUpperCase(), rte);
+        } catch (Exception e) {
+            logger.debug("Foobot device doesn't provide sensor data for channel id = {}: {}", channelId.toUpperCase(),
+                    e);
+        }
+
+        if (value == null) {
+            updateState(channelId, UnDefType.NULL);
+            logger.debug("Update channel {}: null value for the channel", channelId);
+        } else if (value instanceof BigDecimal) {
```

Comment:
@htreu / @wborn I would expect some more logic here to handle UoM or is it so easy?

Reply:
The idea of QuantityType is it can handle different variants automatically. So for example if the user wants the temperature in Fahrenheit this can be specified in the label the users specifies in the configuration. But the binding code must specify what the unit is the that is obtained from the data. The unit specified in the thing-types is not used for this so this must be specified in code. Thus each value must be created with `new QuantityType()` with either 1 argument the value  + unit as string or as 2 arguments: value and unit type.  That can probably be implemented similar to SENSOR_MAP but than returns the unit class (maybe create an enum these sensor types putting everything nicely together.